frontend/src/container/OnboardingContainer/Steps/EnvironmentDetails/EnvironmentDetails.tsx

```diff
@@ -33,16 +38,78 @@ const supportedEnvironments: SupportedEnvironmentsProps[] = [
 ];
 
 export default function EnvironmentDetails(): JSX.Element {
+	const [form] = Form.useForm();
+	const { t } = useTranslation(['common']);
+
 	const {
 		selectedEnvironment,
 		updateSelectedEnvironment,
 		selectedModule,
+		selectedDataSource,
+		selectedFramework,
 		errorDetails,
 		updateErrorDetails,
 	} = useOnboardingContext();
 
+	const requestedEnvironmentName = Form.useWatch(
+		'requestedEnvironmentName',
+		form,
+	);
+
+	const { notifications } = useNotifications();
+
+	const [
+		isSubmittingRequestForEnvironment,
+		setIsSubmittingRequestForEnvironment,
+	] = useState(false);
+
+	const handleRequestedEnvironmentSubmit = async (): Promise<void> => {
+		try {
+			setIsSubmittingRequestForEnvironment(true);
+			const response = await logEvent('Onboarding V2: Environment Requested', {
+				module: selectedModule?.id,
+				dataSource: selectedDataSource?.id,
+				framework: selectedFramework,
+				environment: requestedEnvironmentName,
+			});
+
+			if (response.statusCode === 200) {
+				notifications.success({
+					message: 'Environment Request Submitted',
+				});
+
+				form.setFieldValue('requestedEnvironmentName', '');
+
```

Comment:
this will also be avoided by moving the above code to separate component

Reply:
This won't be reusable beyond onboarding. Not much upside to modularising this.